roles/ceph-osd/tasks/scenarios/non-collocated.yml

```diff
@@ -57,14 +65,21 @@
     - containerized_deployment
     - osd_objectstore == 'bluestore'
     - item.0.partitions|length == 0
+  register: osd
 
-- name: prepare ceph "{{ osd_objectstore }}" non-containerized osd disk(s) non-collocated
-  command: "ceph-disk prepare {{ ceph_disk_cli_options }} {{ item.1 }} {{ item.2 }}"
-  with_together:
-    - "{{ parted_results.results | default([]) }}"
-    - "{{ devices }}"
-    - "{{ dedicated_devices }}"
-  changed_when: false
+- set_fact: osd_created=True
+  when: osd.changed
+
+- block:
+  - name: prepare ceph "{{ osd_objectstore }}" non-containerized osd disk(s) non-collocated
+    command: "ceph-disk prepare {{ ceph_disk_cli_options }} {{ item.1 }} {{ item.2 }}"
+    with_together:
+      - "{{ parted_results.results | default([]) }}"
+      - "{{ devices }}"
+      - "{{ dedicated_devices }}"
+    changed_when: false
```

Comment:
Doesn't this break the register line 99? https://github.com/ceph/ceph-ansible/pull/2599/files#diff-bc1034a6fab2180f375254c274e02ed6R99

Reply:
Can you elaborate on your comment some more?  I'm not sure what you think may not work with the register on 99.

Reply:
@leseb are you concerned because of the `changed_when: false` because it's changing the "changed" attribute of a registered variable ? If yes, the register L99 isn't affected since they are not applied on the same task.

Reply:
Yeah my bad, missed the indent and the block :). Not an issue.